cognite/client/data_classes/_base.py

```diff
@@ -281,14 +280,31 @@ def get(self, id: Optional[int] = None, external_id: Optional[str] = None) -> Op
             return self._id_to_item.get(id)
         return self._external_id_to_item.get(external_id)
 
-    def to_pandas(self, camel_case: bool = False) -> pandas.DataFrame:
+    def to_pandas(
+        self,
+        camel_case: bool = False,
+        expand_metadata: bool = False,
+        metadata_prefix: str = "metadata.",
+    ) -> pandas.DataFrame:
         """Convert the instance into a pandas DataFrame.
 
+        Args:
+            camel_case (bool): Convert column names to camel case (e.g. `externalId` instead of `external_id`)
+            expand_metadata (bool): Expand the metadata column into separate columns.
+            metadata_prefix (str): Prefix to use for metadata columns.
```

Comment:
Should we add a field to the docstring explaining that this function raises when the metadata column names (after adding prefix) collide with existing columns?

Reply:
Would you still do it if we remove the check as per my suggestion? My thoughts are that if it is not an error raised in the function itself, it should not be mentioned in the docstring. Only in the functions that they are raised in.

Reply:
I do not understand what you are referencing here Hayden. A line in the docstring explaining the possible ValueError is good.

Reply:
In general I agree @doctrino, but maybe consider this example, as this might illustrate better what I am trying to say. Let us say you have a function which calls several other functions. Maybe something like this:
```
def func(...):
    func_a(...)
    func_b(...)
    func_c(...)
```
Let us propose that each of these sub-functions all have their own error-raising capabilities and have been documented which includes documenting the errors and that no error-raising happens in `func()`. Would you repeat those errors in the docstring for `func()`? My thoughts are no, because they should already be documented in `func_a`, `func_b`, and `func_c`. @haakonvt wanted to change it such that we use the join instead, which would make this check obsolete since this type of check is already performed in the `join` method. At the end of the day, I will happily update the docs to include it if a majority wants it, but I just don't see the the point if we are using another function (join) which checks the error and no error raising happens in the one being written (to_pandas). 😄

Reply:
Discussion was resolved on Slack. We decided to add a comment in the docstrings as this would be useful for end users (especially new end users).